crypto/bio/bss_conn.c

```diff
@@ -188,8 +188,17 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
 
         case BIO_CONN_S_BLOCKED_CONNECT:
             i = BIO_sock_error(b->num);
-            if (i) {
+            if (i != 0) {
                 BIO_clear_retry_flags(b);
+                if ((c->addr_iter = BIO_ADDRINFO_next(c->addr_iter)) != NULL) {
+                    /*
+                     * if there are more addresses to try, do that first
+                     */
+                    BIO_closesocket(b->num);
```

Comment:
This is copied from above.
b->num should be initialized to INVALID_SOCKET.
since it takes a risk that the socket is close twice. 
I think both places should clear that value since the
BIO_socket could fail, and the following BIO_reset
will close the socket a second time.

Reply:
We need either new PR or add it to #12017